tangorb/dao/src/main/java/fr/esrf/TangoApi/DatabaseDAODefaultImpl.java

```diff
@@ -513,11 +516,11 @@ public void unexport_device(Database database, String devname)
 	 * @see fr.esrf.TangoApi.IDatabaseDAO#export_device(fr.esrf.TangoApi.DbDevExportInfo)
 	 */
     //==========================================================================
-    public void export_device(Database database, DbDevExportInfo devinfo)
+    public void export_device(Database database, DeviceExportInfo deviceExportInfo)
```

Comment:
Hi Igor,
Thank you very much for working on this issue.
If the interface is changed, backwards-compatibility is broken. Are you sure this is what you want to do?
For instance, this method is called directly by Astor. So current versions of Astor won't be compatible with this version.
I tried it and I got the following error when trying to start a server from Astor:
java.lang.NoSuchMethodError: fr.esrf.TangoApi.Database.export_device(Lfr/esrf/TangoApi/DbDevExportInfo;)V
	at admin.astor.TangoHost.registerServer(TangoHost.java:318)
	at admin.astor.HostInfoDialog$StartServersThread.run(HostInfoDialog.java:554)
Astor does not need to specify the PID when using export_device. I think the PID should be set to "null" string in that case.
I think a simple fix would be to change DbDevExportInfo.toStringArray() method in order to create a string array with the parameters in the correct order and the parameter corresponding to PID could be forced to "null" string since it is actually never used by the users of this class (and would not be relevant because the users of this class are mostly standalone clients and you don't want to store the PID of the client).